src/golang/lib/job/databricks.go

```diff
@@ -171,3 +112,118 @@ func (j *DatabricksJobManager) EditCronJob(ctx context.Context, name string, cro
 func (j *DatabricksJobManager) DeleteCronJob(ctx context.Context, name string) JobError {
 	return nil
 }
+
+func (j *DatabricksJobManager) CreateTask(
+	ctx context.Context,
+	workflowName string,
+	spec Spec,
+	parentOperatorNames []string,
+) (*jobs.JobTaskSettings, error) {
+
+	// Get the entrypoint file for Databricks.
+	storageConfig, err := spec.GetStorageConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "Spec unexpectedly has no storage config.")
+	}
+	scriptFile, specStr, err := j.mapJobTypeToFile(spec)
+	if err != nil {
+		return nil, err
+	}
+	bucket := storageConfig.S3Config.Bucket
+	pythonFilePath := fmt.Sprintf("%s/%s", bucket, scriptFile)
+
+	task, err := databricks_lib.CreateTask(
+		ctx,
+		j.databricksClient,
+		workflowName,
+		spec.JobName(),
+		parentOperatorNames,
+		pythonFilePath,
+		specStr,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return task, nil
+}
+
+func (j *DatabricksJobManager) LaunchMutlipleTaskJob(
+	ctx context.Context,
+	name string,
+	taskList []jobs.JobTaskSettings,
+) (int64, JobError) {
+
+	// Create and register the job with Databricks.
+	jobID, err := databricks_lib.CreateJob(ctx, j.databricksClient, name, j.conf.S3InstanceProfileARN, taskList)
+	if err != nil {
+		return -1, systemError(errors.Wrap(err, "Error creating job in Databricks."))
+	}
+	// Trigger the job to execute asynchronously.
+	runID, err := databricks_lib.RunNow(ctx, j.databricksClient, jobID)
+	if err != nil {
+		return -1, systemError(errors.Wrap(err, "Error runnning job in Databricks."))
+	}
+	j.runMap[name] = runID
+	// Get the runIDs of the individual tasks for polling purposes.
+	taskMap, err := databricks_lib.GetTaskRunIDs(ctx, j.databricksClient, runID)
+	if err != nil {
+		return -1, systemError(errors.Wrap(err, "Error creating the task id map."))
+	}
+	for taskName, taskID := range taskMap {
+		j.runMap[taskName] = taskID
+	}
+	return runID, nil
+}
+
+func (j *DatabricksJobManager) mapJobTypeToFile(spec Spec) (string, string, error) {
+	// Add S3 Access Keys to all specs
+	storageConfig, err := spec.GetStorageConfig()
+	if err != nil {
+		return "", "", errors.Wrap(err, "Spec unexpectedly has no storage config.")
+	}
+	storageConfig.S3Config.AWSAccessKeyID = j.conf.AwsAccessKeyID
+	storageConfig.S3Config.AWSSecretAccessKey = j.conf.AwsSecretAccessKey
+	if spec.Type() == FunctionJobType {
+		functionSpec, ok := spec.(*FunctionSpec)
+		if !ok {
+			return "", "", ErrInvalidJobSpec
+		}
+
+		functionSpec.FunctionExtractPath = defaultFunctionExtractPath
+		specStr, err := EncodeSpec(spec, JsonSerializationType)
+		if err != nil {
+			return "", "", errors.Wrap(err, "Unable to encode spec.")
+		}
+		return databricks_lib.DatabricksFunctionScript, specStr, nil
+
+	} else if spec.Type() == ParamJobType {
+		specStr, err := EncodeSpec(spec, JsonSerializationType)
+		if err != nil {
+			return "", "", errors.Wrap(err, "Unable to encode spec.")
+		}
+		return databricks_lib.DatabricksParamScript, specStr, nil
```

Comment:
are these 'Script' file names?

Reply:
Yes, these are the entrypoint file names used by Databricks Jobs. These files are uploaded to S3 when a user connects to a Databricks integration!